Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -0,0 +1,869 @@
+    auto* load_btn = new QPushButton(tr("Load"));
+
+    connect(clear_btn, &QAbstractButton::clicked, this, [this, i]() { ClearSkylander(i); });
+    connect(create_btn, &QAbstractButton::clicked, this, [this, i]() { CreateSkylander(i); });
+    connect(load_btn, &QAbstractButton::clicked, this, [this, i]() { LoadSkylander(i); });
+
+    hbox_skylander->addWidget(label_skyname);
+    hbox_skylander->addWidget(m_edit_skylanders[i]);
+    hbox_skylander->addWidget(clear_btn);
+    hbox_skylander->addWidget(create_btn);
+    hbox_skylander->addWidget(load_btn);
+
+    vbox_group->addLayout(hbox_skylander);
+  }
+
+  m_group_skylanders->setLayout(vbox_group);
+  scroll_area->setWidget(m_group_skylanders);
+  scroll_area->setWidgetResizable(true);
+  m_group_skylanders->setVisible(Config::Get(Config::MAIN_EMULATE_SKYLANDER_PORTAL));
+  main_layout->addWidget(scroll_area);
+  setLayout(main_layout);
+
+  UpdateEdits();
+}
+
+void SkylanderPortalWindow::OnEmulationStateChanged(Core::State state)
+{
+  const bool running = state != Core::State::Uninitialized;
+
+  m_checkbox->setEnabled(!running);
+}
+
+CreateSkylanderDialog::CreateSkylanderDialog(QWidget* parent) : QDialog(parent)
+{
+  setWindowTitle(tr("Skylander Creator"));
+  setObjectName(QString::fromStdString("skylanders_creator"));
+  setMinimumSize(QSize(500, 150));
+  auto* layout = new QVBoxLayout;
+
+  auto* combo_skylist = new QComboBox();
+  QStringList filterlist;
+  for (const auto& entry : list_skylanders)
+  {
+    const uint qvar = (entry.first.first << 16) | entry.first.second;
+    combo_skylist->addItem(QString::fromStdString(entry.second), QVariant(qvar));
+    filterlist << QString::fromStdString(entry.second);
+  }
+  combo_skylist->addItem(tr("--Unknown--"), QVariant(0xFFFFFFFF));
+  combo_skylist->setEditable(true);
+  combo_skylist->setInsertPolicy(QComboBox::NoInsert);
+
+  auto* co_compl = new QCompleter(filterlist, this);
+  co_compl->setCaseSensitivity(Qt::CaseInsensitive);
+  co_compl->setCompletionMode(QCompleter::PopupCompletion);
+  co_compl->setFilterMode(Qt::MatchContains);
+  combo_skylist->setCompleter(co_compl);
+
+  layout->addWidget(combo_skylist);
+
+  auto* line = new QFrame();
+  line->setFrameShape(QFrame::HLine);
+  line->setFrameShadow(QFrame::Sunken);
+  layout->addWidget(line);
+
+  auto* hbox_idvar = new QHBoxLayout();
+  auto* label_id = new QLabel(tr("ID:"));
+  auto* label_var = new QLabel(tr("Variant:"));
+  auto* edit_id = new QLineEdit(QString::fromStdString("0"));
+  auto* edit_var = new QLineEdit(QString::fromStdString("0"));
+  auto* rxv =
+      new QRegularExpressionValidator(QRegularExpression(QString::fromStdString("\\d*")), this);
+  edit_id->setValidator(rxv);
+  edit_var->setValidator(rxv);
+  hbox_idvar->addWidget(label_id);
+  hbox_idvar->addWidget(edit_id);
+  hbox_idvar->addWidget(label_var);
+  hbox_idvar->addWidget(edit_var);
+  layout->addLayout(hbox_idvar);
+
+  auto* hbox_buttons = new QHBoxLayout();
```

Comment:
The Qt docs recommend using a [QDialogButtonBox](https://doc.qt.io/qt-6/qdialogbuttonbox.html) instead.

> Dialogs and message boxes typically present buttons in a layout that conforms to the interface guidelines for that platform. Invariably, different platforms have different layouts for their dialogs. QDialogButtonBox allows a developer to add buttons to it and will automatically use the appropriate layout for the user's desktop environment.

```cpp
    auto* buttons = new QDialogButtonBox(QDialogButtonBox::Ok | QDialogButtonBox::Cancel);
    buttons->button(QDialogButtonBox::Ok)->setText(tr("Create"));
    connect(buttons, &QDialogButtonBox::accepted, this, ...);
    connect(buttons, &QDialogButtonBox::rejected, this, ...);
```